api/v1alpha1/validation/securitypolicy_validate.go

```diff
@@ -1,154 +0,0 @@
-// Copyright Envoy Gateway Authors
-// SPDX-License-Identifier: Apache-2.0
-// The full text of the Apache license is available in the LICENSE file at
-// the root of the repo.
-
-package validation
-
-import (
-	"errors"
-	"fmt"
-	"net/mail"
-	"net/url"
-	"strings"
-
-	utilerrors "k8s.io/apimachinery/pkg/util/errors"
-	"k8s.io/apimachinery/pkg/util/validation"
-
-	egv1a1 "github.com/envoyproxy/gateway/api/v1alpha1"
-)
-
-// ValidateSecurityPolicy validates the provided SecurityPolicy.
-func ValidateSecurityPolicy(policy *egv1a1.SecurityPolicy) error {
-	var errs []error
-	if policy == nil {
-		return errors.New("policy is nil")
-	}
-	if err := validateSecurityPolicySpec(&policy.Spec); err != nil {
-		errs = append(errs, err)
-	}
-
-	return utilerrors.NewAggregate(errs)
-}
-
-// validateSecurityPolicySpec validates the provided spec.
-func validateSecurityPolicySpec(spec *egv1a1.SecurityPolicySpec) error {
-	var errs []error
-
-	sum := 0
-	switch {
-	case spec == nil:
-		errs = append(errs, errors.New("spec is nil"))
-	case spec.CORS != nil:
-		sum++
-	case spec.JWT != nil:
-		sum++
-		if err := ValidateJWTProvider(spec.JWT.Providers); err != nil {
-			errs = append(errs, err)
-		}
-	case spec.Authorization != nil:
-		sum++
-	case spec.APIKeyAuth != nil:
-		sum++
-	case spec.BasicAuth != nil:
-		sum++
-	case spec.ExtAuth != nil:
-		sum++
-	case spec.OIDC != nil:
-		sum++
```

Comment:
Remove this empty spec check because other policy  types don't do this check. We can apply it to all the policy types in a follow-up PR if it's needed.